### PR TITLE
Custom message cog

### LINKED ISSUE #181: [FEAT] Custom message cog
# Overview

A cog that allows moderators to send and edit messages using the bot. This could be useful for announcements, rules and the like - as it allows all moderators to edit messages, rather than relying on the person who originally sent the message to update details. Ideally this should support embeds, and the process of inputting content should be interactive.

# Commands
```
[p]msg create <#channel>
[p]msg edit <message>
```

# DoD
- [x] `create` and `edit` commands that can be used by all mods
- [x] Support raw messages and embeds

# Stretch
- [x] Support messages with an embed and raw content

### PR DESCRIPTION
**Pull request by issy**
# Initial Checklist
- [x] Has @tigattack / @issy been added as a reviewer?
- [x] If applicable, have the relevant project(s), milestone(s), and label(s) been applied?
- [x] If applicable, have you added details of the cog to the readme as per [README.md](https://github.com/rHomelab/LabBot-Cogs/blob/develop/README.md#cog-summaries)?

# Details

This PR adds an interactive message builder for moderators to send messages using the bot.
This can be seen in action here.
image

image

* **Does this resolve an issue?**
Resolves #181

### Features / Fixes
* Adds `[p]msg create` command to send messages using the bot
* Adds `[p]msg edit` command to edit messages sent by the bot

### Breaking Changes
No breaking changes

## Additional
The current `[p]msg edit` implementation will overwrite the unspecified fields. In other words, if you want to edit a message with text and an embed, and you only want to edit the embed, you will have to choose the `both` option and input the current text body, otherwise it will be removed. This was a stylistic choice, and can be removed if there are any disagreements. The benefit of this method is that you can remove elements from a message (content or embed) if you want to.
EDIT: This is no longer the case - the bot will now ask you if you want to remove existing fields from the message if you have not specified them in your edit session.

### CODE REVIEW

**Comment by tigattack** [comment]
A comment on your final note: How about if the cog detected if the message being edited contains both elements and prompt if you wish to remove one. Then, if one is removed, whether you wish to edit the other, else both remain and you are prompted which (or both) to edit.

This is just an idea I had and I'm unsure as to difficulty of implementation. I don't to cause a bunch more work for you if it'd be too much. This already looks awesome and the functionality as-is is fine, just needs to be clear in the docs/help texts.

**Comment by issy** [comment]
It turns out this last change is a lot of work, but I'm already like halfway through it, so I may as well complete it

**Comment by tigattack** [comment]
Much appreciated mate 🙏

**Comment by issy** [comment]
Ready for review again
image
image

File: README.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ Cogs for the [RED](https://github.com/Cog-Creators/Red-DiscordBot/)-based [Homel
   - [AutoReact](#autoreact)
   - [AutoReply](#autoreply)
   - [Convert](#convert)
+  - [Custom msg](#custom-msg)
```
Comment (tigattack) [inline README.md:20]:
```suggestion
  - [Custom-msg](#custom-msg)
```

For consistent naming

File: custom_msg/info.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+    "author": [
+        "Issy"
+    ],
+    "short": "",
+    "description": "",
```
Comment (tigattack) [inline custom_msg/info.json:6]:
```suggestion
    "short": "Send messages as the bot",
    "description": "Allows moderators to send messages as the bot, also allowing modification of existing messages sent from the cog",

```

Cog short & description.

**Review by tigattack — CHANGES_REQUESTED** [review]
Amazing, thank you Issy 🙏 As with #191, can't really provide a proper review but what I understand looks ace.

Couple of suggestions for docs/info but otherwise let's goooo.

**Review by tigattack — APPROVED** [review]

**Pull request merged**